diff -aur a/src/pybind/ceph_argparse.py b/src/pybind/ceph_argparse.py
--- a/src/pybind/ceph_argparse.py	2020-01-09 05:36:54.000000000 +1100
+++ b/src/pybind/ceph_argparse.py	2020-02-14 13:48:54.198934108 +1100
@@ -1299,7 +1299,7 @@
     if timeout == 0 or timeout == None:
         # python threading module will just get blocked if timeout is `None`,
         # otherwise it will keep polling until timeout or thread stops.
-        timeout = 2 ** 32
+        timeout = (1 << (32 - 1)) - 1
     t = RadosThread(func, *args, **kwargs)
 
     # allow the main thread to exit (presumably, avoid a join() on this
