--- a/src/os/bluestore/BlueFS.h	2019-04-26 04:15:48.000000000 +1000
+++ b/src/os/bluestore/BlueFS.h	2019-07-08 12:35:44.115849178 +1000
@@ -359,7 +359,7 @@
   int _read_random(
     FileReader *h,   ///< [in] read from here
     uint64_t offset, ///< [in] offset
-    size_t len,      ///< [in] this many bytes
+    uint64_t len,    ///< [in] this many bytes
     char *out);      ///< [out] optional: or copy it here
 
   void _invalidate_cache(FileRef f, uint64_t offset, uint64_t length);
diff -Naur a/src/osd/osd_types.h b/src/osd/osd_types.h
