diff -aur a/src/dmclock/support/src/indirect_intrusive_heap.h b/src/dmclock/support/src/indirect_intrusive_heap.h
--- a/src/dmclock/support/src/indirect_intrusive_heap.h	2019-12-19 09:15:27.000000000 +1100
+++ b/src/dmclock/support/src/indirect_intrusive_heap.h	2020-04-08 23:18:35.102087776 +1000
@@ -48,7 +48,7 @@
 	   typename T,
 	   IndIntruHeapData T::*heap_info,
 	   typename C,
-	   uint K = 2>
+	   unsigned int K = 2>
   class IndIntruHeap {
 
     // shorthand
