--- a/src/osdc/Striper.cc	2019-04-26 04:15:48.000000000 +1000
+++ b/src/osdc/Striper.cc	2019-07-08 12:37:23.737475828 +1000
@@ -328,7 +328,7 @@
 
       ceph_assert(s->first <= bl_off);
       size_t left = (s->first + s->second) - bl_off;
-      size_t actual = std::min(left, tlen);
+      size_t actual = std::min<size_t>(left, tlen);
 
       if (actual > 0) {
 	ldout(cct, 20) << "  s has " << actual << ", copying" << dendl;
