diff -aur a/build/makelib/golang.mk b/build/makelib/golang.mk
--- a/build/makelib/golang.mk	2020-01-07 05:17:07.000000000 +1100
+++ b/build/makelib/golang.mk	2020-01-12 23:56:01.375199447 +1100
@@ -195,7 +195,7 @@
 $(DEP):
 	@echo === installing dep
 	@mkdir -p $(TOOLS_HOST_DIR)/tmp
-	@if [ "$(GOHOSTARCH)" = "arm64" ]; then\
+	@if [[ "$(GOHOSTARCH)" == "arm"* ]]; then\
 		GOPATH=$(TOOLS_HOST_DIR)/tmp GOBIN=$(TOOLS_HOST_DIR) $(GOHOST) get -u github.com/golang/dep/cmd/dep;\
 		mv $(TOOLS_HOST_DIR)/dep $@;\
 	else \
