--- a/src/os/bluestore/BlueFS.cc	2019-04-26 04:15:48.000000000 +1000
+++ b/src/os/bluestore/BlueFS.cc	2019-07-08 12:38:42.380865732 +1000
@@ -1305,7 +1305,7 @@
 int BlueFS::_read_random(
   FileReader *h,         ///< [in] read from here
   uint64_t off,          ///< [in] offset
-  size_t len,            ///< [in] this many bytes
+  uint64_t len,          ///< [in] this many bytes
   char *out)             ///< [out] optional: or copy it here
 {
   dout(10) << __func__ << " h " << h
diff -Naur a/src/os/bluestore/BlueFS.h b/src/os/bluestore/BlueFS.h
