--- a/src/include/buffer.h	2019-04-26 04:15:48.000000000 +1000
+++ ceph-14.2.1/src/include/buffer.h	2019-07-08 18:21:42.389192704 +1000
@@ -735,9 +735,7 @@
 	//return off == bl->length();
       }
 
-      void advance(int o) = delete;
       void advance(unsigned o);
-      void advance(size_t o) { advance(static_cast<unsigned>(o)); }
       void seek(unsigned o);
       char operator*() const;
       iterator_impl& operator++();
