--- a/src/dmclock/src/dmclock_util.cc	2019-02-28 13:45:31.000000000 +1100
+++ ceph-14.2.1/src/dmclock/src/dmclock_util.cc	2019-07-08 20:29:00.721144739 +1000
@@ -21,7 +21,7 @@
 #include "dmclock_util.h"
 
 
-std::string crimson::dmclock::format_time(const Time& time, uint modulo) {
+std::string crimson::dmclock::format_time(const Time& time, unsigned modulo) {
   long subtract = long(time / modulo) * modulo;
   std::stringstream ss;
   ss << std::fixed << std::setprecision(4) << (time - subtract);
diff -Naur a/src/dmclock/src/dmclock_util.h ceph-14.2.1/src/dmclock/src/dmclock_util.h
